pennylane/tape/tape.py

```diff
@@ -930,6 +936,268 @@ def set_parameters(self, params, trainable_only=True):
             op = self._par_info[idx]["op"]
             op.data[self._par_info[idx]["p_idx"]] = p
 
+    @staticmethod
+    def _single_measurement_shape(measurement_process, device):
+        """Auxiliary function of shape that determines the output
+        shape of a tape with a single measurement.
+
+        Args:
+            measurement_process (MeasurementProcess): the measurement process
+                associated with the single measurement
+            device (~.Device): a PennyLane device
+
+        Returns:
+            tuple: output shape
+        """
+        return measurement_process.shape(device)
+
+    @staticmethod
+    def _multi_homogenous_measurement_shape(mps, device):
+        """Auxiliary function of shape that determines the output
+        shape of a tape with multiple homogenous measurements.
+
+        .. note::
+
+            Assuming multiple probability measurements where not all
+            probability measurements have the same number of wires specified,
+            the output shape of the tape is a sum of the output shapes produced
+            by each probability measurement.
+
+            Consider the `qml.probs(wires=[0]), qml.probs(wires=[1,2])`
+            multiple probability measurement with an analytic device as an
+            example.
+
+            The output shape will be a one element tuple `(6,)`, where the
+            element `6` is equal to `2 ** 1 + 2 ** 2 = 6`. The base of each
+            term is determined by the number of basis states and the exponent
+            of each term comes from the length of the wires specified for the
+            probability measurements: `1 == len([0]) and 2 == len([1, 2])`.
+        """
+        shape = tuple()
+
+        # We know that there's one type of return_type, gather it from the
+        # first one
+        ret_type = mps[0].return_type
+        if ret_type == qml.measurements.State:
+            raise TapeError(
```

Comment:
I do not understand why we cannot have this one, is it not something similar to `shape=(len(mps), )`

Reply:
We don't support multiple state measurements in tapes.